base/complex.jl

```diff
@@ -424,7 +424,8 @@ function log2(z::Complex)
     a/log(oftype(real(a),2))
 end
 
-function exp(z::Complex)
+exp(z::Complex) = exp(float(z))
```

Comment:
Ref https://github.com/JuliaLang/julia/pull/13539 for a better workaround to the same issue.